internal/build/phase_config_provider.go

```diff
@@ -94,8 +105,12 @@ func WithBinds(binds ...string) PhaseConfigProviderOperation {
 
 func WithDaemonAccess() PhaseConfigProviderOperation {
 	return func(provider *PhaseConfigProvider) {
-		provider.ctrConf.User = "root"
-		provider.hostConf.Binds = append(provider.hostConf.Binds, "/var/run/docker.sock:/var/run/docker.sock")
+		WithRoot()(provider)
```

Comment:
What's happening here? This is calling `WithRoot`, and naming the return value `provider`?

Reply:
Because `WithRoot()` is a higher-order function and returns a function, we have to then call that returned function to perform its behavior.